Sources/StreamChat/WebSocketClient/Events/MemberEvents_Tests.swift

```diff
@@ -29,3 +30,97 @@ class MemberEvents_Tests: XCTestCase {
+    var client: ChatClient!
+    var currentUserId: UserId!
+
+    let eventDecoder = EventDecoder<NoExtraData>()
+
+    override func setUp() {
+        super.setUp()
+
+        var config = ChatClientConfig(apiKeyString: "Integration_Tests_Key")
+        config.isLocalStorageEnabled = false
+        config.isClientInActiveMode = false
+        
+        currentUserId = .unique
+        client = ChatClient(config: config, tokenProvider: .development(userId: currentUserId))
+        try! client.databaseContainer.createCurrentUser(id: currentUserId)
+        client.eventNotificationCenter.eventBatchPeriod = 0
+    }
+
+    func test_MemberAddedEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "MemberAdded")
+        let event = try eventDecoder.decode(from: json) as? MemberAddedEvent
+
+        client.eventNotificationCenter.process(event!)
+
+        AssertAsync {
+            Assert.willNotBeNil(
+                self.client.databaseContainer.viewContext.member(
+                    userId: "steep-moon-9",
+                    cid: ChannelId(type: .messaging, id: "new_channel_9125")
+                )
+            )
+        }
+    }
+    
+    func test_MemberUpdatedEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "MemberUpdated")
+        let event = try eventDecoder.decode(from: json) as? MemberUpdatedEvent
+
+        client.eventNotificationCenter.process(event!)
+
+        AssertAsync {
+            Assert.willNotBeNil(
+                self.client.databaseContainer.viewContext.member(
+                    userId: "count_dooku",
+                    cid: ChannelId(type: .messaging, id: "!members-jkE22mnWM5tjzHPBurvjoVz0spuz4FULak93veyK0lY")
+                )
+            )
+        }
+    }
+    
+    func test_MemberRemovedEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "MemberRemoved")
+        let event = try eventDecoder.decode(from: json) as? MemberRemovedEvent
+                
+        let channelId = ChannelId(type: .messaging, id: "!members-jkE22mnWM5tjzHPBurvjoVz0spuz4FULak93veyK0lY")
+        
+        // First create channel and member of that channel to be saved in database.
+        try! client.databaseContainer.createChannel(
+            cid: channelId,
+            withMessages: false,
+            withQuery: false
+        )
+        
+        try! client.databaseContainer.createMember(
+            userId: "r2-d2",
+            role: .member,
+            cid: ChannelId(type: .messaging, id: "!members-jkE22mnWM5tjzHPBurvjoVz0spuz4FULak93veyK0lY"),
+            query: nil
+        )
+        
+        // Check if those are created in order to avoid false-positive.
+        XCTAssertNotNil(
+            client.databaseContainer.viewContext.member(
+                userId: "r2-d2",
+                cid: ChannelId(type: .messaging, id: "!members-jkE22mnWM5tjzHPBurvjoVz0spuz4FULak93veyK0lY")
+            )
+        )
```

Comment:
I'd do something like:
```
XCTAssertTrue(channel.members.contains { $0.user.id == <id> })
// run the event
AssertAsync.willBeFalse(channel.members.contains { $0.user.id == <id> })
```
and that's it :) No need to check the counts and the current user